contracts/verification/OraclizeVerifier.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity ^0.4.11;
+
+import "./Verifier.sol";
+import "./Verifiable.sol";
+
+import "../../installed_contracts/oraclize/contracts/usingOraclize.sol";
+
+/*
+ * @title Verifier contract that uses Oraclize for off-chain computation
+ */
+contract OraclizeVerifier is Verifier, usingOraclize {
+
+    // Stores parameters for an Oraclize query
+    struct OraclizeQuery {
+        uint256 jobId;
+        uint256 segmentSequenceNumber;
+        address callbackContract;
+    }
+
+    // Stores active Oraclize queries
+    mapping (bytes32 => OraclizeQuery) oraclizeQueries;
+
+    // Check if sender is Oraclize
+    modifier onlyOraclize() {
+        if (msg.sender != oraclize_cbAddress()) throw;
+        _;
+    }
+
+    // Check if sufficient funds for Oraclize computation
+    modifier sufficientOraclizeFunds() {
+        if (oraclize_getPrice("computation") > this.balance) throw;
```

Comment:
Think we should check against this.balance or msg.amount? I'm wondering if there's a scenario where one transaction could use funds from another previous transaction without paying if this contract continued to accumulate balance somehow.